internal/resources/metal/vlans/resource.go

```diff
@@ -0,0 +1,186 @@
+		return
+	}
+	if !(data.Facility.IsNull() && data.Vxlan.IsNull()) {
+		response.Diagnostics.AddError("Invalid inout params",
+			equinix_errors.FriendlyError(errors.New("you can set vxlan only for metro vlans")).Error())
+		return
+	}
+
+	createRequest := &packngo.VirtualNetworkCreateRequest{
+		ProjectID:   data.ProjectID.ValueString(),
+		Description: data.Description.ValueString(),
+	}
+	if !data.Metro.IsNull() {
+		createRequest.Metro = data.Metro.ValueString()
+		createRequest.VXLAN = int(data.Vxlan.ValueInt64())
+	}
+	if !data.Facility.IsNull() {
+		createRequest.Facility = data.Facility.ValueString()
+	}
+	vlan, _, err := client.ProjectVirtualNetworks.Create(createRequest)
+	if err != nil {
+		response.Diagnostics.AddError("Error creating Vlan", equinix_errors.FriendlyError(err).Error())
+		return
+	}
+
+	// Parse API response into the Terraform state
+	response.Diagnostics.Append(data.parse(vlan)...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	// Set state to fully populated data
+	response.Diagnostics.Append(response.State.Set(ctx, &data)...)
+	return
+}
+
+func (r *Resource) Read(ctx context.Context, request resource.ReadRequest, response *resource.ReadResponse) {
+	r.Meta.AddFwModuleToMetalUserAgent(ctx, request.ProviderMeta)
+	client := r.Meta.Metal
+
+	var data ResourceModel
+	response.Diagnostics.Append(request.State.Get(ctx, &data)...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	vlan, _, err := client.ProjectVirtualNetworks.Get(
+		data.ID.ValueString(),
+		&packngo.GetOptions{Includes: []string{"assigned_to"}},
+	)
+	if err != nil {
+		if equinix_errors.IsNotFound(err) {
+			response.Diagnostics.AddWarning(
+				"Equinix Metal Vlan not found during refresh",
+				fmt.Sprintf("[WARN] Vlan (%s) not found, removing from state", data.ID.ValueString()),
+			)
+			response.State.RemoveResource(ctx)
+			return
+		}
+		response.Diagnostics.AddError("Error fetching Vlan using vlanId",
+			equinix_errors.FriendlyError(err).Error())
+		return
+	}
+
+	response.Diagnostics.Append(data.parse(vlan)...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	response.Diagnostics.Append(response.State.Set(ctx, &data)...)
+}
+
+func (r *Resource) Update(ctx context.Context, request resource.UpdateRequest, response *resource.UpdateResponse) {
+	//TODO implement me
+	panic("implement me")
+}
+
+func (r *Resource) Delete(ctx context.Context, request resource.DeleteRequest, response *resource.DeleteResponse) {
+	r.Meta.AddFwModuleToMetalUserAgent(ctx, request.ProviderMeta)
+	client := r.Meta.Metal
+
+	var data ResourceModel
+	response.Diagnostics.Append(request.State.Get(ctx, &data)...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	vlan, resp, err := client.ProjectVirtualNetworks.Get(
+		data.ID.ValueString(),
+		&packngo.GetOptions{Includes: []string{"instances", "meta_gateway"}},
+	)
+	if err != nil {
+		if resp.StatusCode != http.StatusForbidden && resp.StatusCode != http.StatusNotFound {
```

Comment:
While this block structure change is, as far as I can tell, equivalent to the original, changing things like this introduces unnecessary risk.  Is there an incompatibility between framework code and the `IgnoreResponseErrors` function?

Reply:
I checked and functionally there are no differences. Still I reused `IgnoreResponseErrors`